server/src-lib/Hasura/Events/Lib.hs

```diff
@@ -169,20 +177,20 @@ retryAfterHeader = "Retry-After"
 initEventEngineCtx :: Int -> DiffTime -> STM EventEngineCtx
 initEventEngineCtx maxT _eeCtxFetchInterval = do
   _eeCtxEventThreadsCapacity <- newTVar maxT
+  _eeCtxLockedEvents <- newTVar Set.empty
   return $ EventEngineCtx{..}
 
 -- | Service events from our in-DB queue.
 --
 -- There are a few competing concerns and constraints here; we want to...
 --   - fetch events in batches for lower DB pressure
 --   - don't fetch more than N at a time (since that can mean: space leak, less
---     effective scale out, possible double sends for events we've checked out 
---     on exit (TODO clean shutdown procedure))
+--     effective scale out
```

Comment:
Hm I guess this TODO isn't actually done here if we continue to unlock all events everytime an HGE comes online...

Reply:
Ah yes, I removed it when I was not unlocking all the events at the startup, I'll add it again